controller/controlplane/controller.go

```diff
@@ -353,6 +353,11 @@ func (r *Reconciler) constructControlPlaneManagerConfigOptions(
 		return nil, fmt.Errorf("failed to get client key from mTLS secret %s", client.ObjectKeyFromObject(mtlsSecret))
 	}
 
+	payloadCustomizer, err := defaultPayloadCustomizer(nil)
```

Comment:
Any chance to make this not require passing an explicit `nil` in non-test code?

We can make that with some functional options I guess (which amend a config, internal to the `defaultPayloadCustomizer`. WDYT?

Reply:
Done, even though I think it's overkill.